test/types/mongodb.test-d.ts

```diff
@@ -38,6 +40,14 @@ const composedMap = mappedAgg.map<string>(x => x.toString());
 expectType<AggregationCursor<string>>(composedMap);
 expectType<string | null>(await composedMap.next());
 expectType<string[]>(await composedMap.toArray());
+expectType<ClientSession>(
```

Comment:
Could we move these into a root level `test/types/sessions.test-d.ts`, adding a comment above along the lines of `// should allow ReadConcern or ReadConcernLike as readConcern in defaultTransactionOptions`, and then also add a sanity test like `expectError(client.startSession({ ... { readConcern: <some invalid parameter> } }))`

Reply:
Moved to new root test in `test/types/sessions.test-d.ts` and added error check.